resilience4j-core/src/test/java/io/github/resilience4j/core/EventProcessorTest.java

```diff
@@ -107,4 +110,48 @@ public void testNoConsumers() {
         assertThat(consumed).isEqualTo(false);
     }
 
+
+    @Test
+    public void testOnEventParallel() throws ExecutionException {
+        EventProcessor<Number> eventProcessor = new EventProcessor<>();
+        EventConsumer<Integer> eventConsumer1 = event -> {
+            // Artificial delay in a listener's code
+            System.out.println("1" + "start" + event.toString());
+            try {
+                Thread.sleep(10 * 1000L);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            System.out.println("1" + "end" + event.toString());
+        };
+
+        EventConsumer<Integer> eventConsumer2 = event -> System.out.println("2" + event.toString());
+
+        // 1st consumer is added
+        eventProcessor.registerConsumer(Integer.class.getSimpleName(), eventConsumer1);
+
+        // process first event in a separate thread to create a race condition
+        CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
+            eventProcessor.processEvent(1); // sleeps 10 sec inside listener's code
+        });
+
+        try {
+            Thread.sleep(1 * 1000L);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+
+        // 2nd consumer is added
+        eventProcessor.registerConsumer(Integer.class.getSimpleName(), eventConsumer2);
+
+        try {
+            future.get();
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        } catch (ExecutionException e) {
+            e.printStackTrace();
+            throw e;
+        }
+    }
```

Comment:
```suggestion
    public void testOnEventParallel() throws ExecutionException, InterruptedException {
        CountDownLatch eventConsumed = new CountDownLatch(1);
        CountDownLatch waitForConsumerRegistration = new CountDownLatch(1);

        EventProcessor<Number> eventProcessor = new EventProcessor<>();
        EventConsumer<Integer> eventConsumer1 = event -> {
            try {
                eventConsumed.countDown();
                waitForConsumerRegistration.await(5, TimeUnit.SECONDS);
                logger.info(event.toString());
            } catch (InterruptedException e) {
                fail("Must not happen");
            }
        };

        EventConsumer<Integer> eventConsumer2 = event -> logger.info(event.toString());

        // 1st consumer is added
        eventProcessor.registerConsumer(Integer.class.getSimpleName(), eventConsumer1);

        // process first event in a separate thread to create a race condition
        CompletableFuture<Void> future = CompletableFuture.runAsync(() -> {
            eventProcessor.processEvent(1); // blocks because of the count down latch
        });

        eventConsumed.await(1, TimeUnit.SECONDS);

        // 2nd consumer is added
        eventProcessor.registerConsumer(Integer.class.getSimpleName(), eventConsumer2);

        future.get();

        waitForConsumerRegistration.countDown();

        then(logger).should(times(1)).info("1");
    }
```